storage/innobase/srv/srv0srv.cc

```diff
@@ -3685,139 +3680,174 @@ srv_enable_redo_encryption_rk(THD* thd)
 /** Enable the undo log encryption if it is set.
 It will try to enable the undo log encryption and write the metadata to
 undo log file header, if innodb_undo_log_encrypt is ON. */
-static
-void
-srv_enable_undo_encryption_if_set()
+bool
+srv_enable_undo_encryption(THD* thd)
 {
 	fil_space_t*	space;
-	const char*	cant_set_undo_tablespace = "Can't set undo tablespace";
-	const char*	to_be_encrypted = " to be encrypted";
 	if (srv_shutdown_state != SRV_SHUTDOWN_NONE) {
-		return;
+		return false;
 	}
 
-	/* Check if encryption for undo log is enabled or not. If it's
-	   enabled, we will store the encryption metadata to the space header
-	   and start to encrypt the undo log block from now on. */
-	if (srv_undo_log_encrypt) {
-		if (srv_undo_tablespaces == 0) {
-			srv_undo_log_encrypt = false;
-			ib::error() << cant_set_undo_tablespace << "s"
-				<< to_be_encrypted
-				<< ", since innodb_undo_tablespaces=0.";
-			return;
-		}
-		if (srv_read_only_mode) {
-			srv_undo_log_encrypt = false;
-			ib::error() << cant_set_undo_tablespace << "s"
-				<< to_be_encrypted
-				<< " in read-only mode.";
-			return;
+	if (srv_undo_tablespaces == 0) {
+		srv_undo_log_encrypt = false;
+		ib::error() << "Can't set undo tablespace" << "s"
+			<< " to be encrypted"
+			<< ", since innodb_undo_tablespaces=0.";
+
+                ib_senderrf(
+                                thd, IB_LOG_LEVEL_WARN,
+                                ER_UNDO_ENCRYPTION_ERROR,
+		                "Can't set undo tablespaces"
+			        " to be encrypted"
+			        ", since innodb_undo_tablespaces=0."
+                );
+
+		return true;
+	}
+	if (srv_read_only_mode) {
+		srv_undo_log_encrypt = false;
+		ib::error() << "Can't set undo tablespace" << "s"
+			<<" to be encrypted" 
+			<< " in read-only mode.";
+                ib_senderrf(
+                                thd, IB_LOG_LEVEL_WARN,
+                                ER_UNDO_ENCRYPTION_ERROR,
+		                "Can't set undo tablespaces"
+			        " to be encrypted"
+			        ", since innodb_undo_tablespaces=0."
```

Comment:
this error message doesn't seem right for read only mode. Also innodb_undo_tablespaces is deprecated and not configurable since 8.0.14 (check https://dev.mysql.com/doc/refman/8.0/en/innodb-undo-tablespaces.html)